### ISSUE TITLE
configuration error: '_________' is not a valid rule identifier

### ISSUE THREAD

**Issue by danieltorrecillas** [realm/SwiftLint#2446]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When swiftlint runs during the build phase in Xcode, two rules that are opted in to in the project's `.swiftlint.yml` are not recognized as valid rules.

The rules are `collection_alignment` and `unused_import`. I noticed that these rules do not show up in the output of `swiftlint rules` though they're documented to be in `Rules.md` and also present in the source code of the Swiftlint project.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Loading configuration from '.swiftlint.yml'
Configuration contains invalid keys:
["analyzer_rules"]
configuration error: 'collection_alignment' is not a valid rule identifier
configuration error: 'unused_import' is not a valid rule identifier
Valid rule identifiers:
anyobject_protocol
array_init
attributes
block_based_kvo
class_delegate_protocol
closing_brace
closure_end_indentation
closure_parameter_position
closure_spacing
colon
comma
compiler_protocol_init
conditional_returns_on_newline
contains_over_first_not_nil
control_statement
convenience_type
custom_rules
cyclomatic_complexity
discarded_notification_center_observer
discouraged_direct_init
discouraged_object_literal
discouraged_optional_boolean
discouraged_optional_collection
dynamic_inline
empty_count
empty_enum_arguments
empty_parameters
empty_parentheses_with_trailing_closure
empty_string
empty_xctest_method
explicit_acl
explicit_enum_raw_value
explicit_init
explicit_top_level_acl
explicit_type_interface
extension_access_modifier
fallthrough
fatal_error_message
file_header
file_length
file_name
first_where
for_where
force_cast
force_try
force_unwrapping
function_body_length
function_default_parameter_at_end
function_parameter_count
generic_type_name
identifier_name
implicit_getter
implicit_return
implicitly_unwrapped_optional
is_disjoint
joined_default_parameter
large_tuple
leading_whitespace
legacy_cggeometry_functions
legacy_constant
legacy_constructor
legacy_nsgeometry_functions
let_var_whitespace
line_length
literal_expression_end_indentation
lower_acl_than_parent
mark
missing_docs
modifier_order
multiline_arguments
multiline_function_chains
multiline_parameters
multiple_closures_with_trailing_closure
nesting
nimble_operator
no_extension_access_modifier
no_fallthrough_only
no_grouping_extension
notification_center_detachment
number_separator
object_literal
opening_brace
operator_usage_whitespace
operator_whitespace
overridden_super_call
override_in_extension
pattern_matching_keywords
prefixed_toplevel_constant
private_action
private_outlet
private_over_fileprivate
private_unit_test
prohibited_super_call
protocol_property_accessors_order
quick_discouraged_call
quick_discouraged_focused_test
quick_discouraged_pending_test
redundant_discardable_let
redundant_nil_coalescing
redundant_optional_initialization
redundant_set_access_control
redundant_string_enum_value
redundant_type_annotation
redundant_void_return
required_enum_case
return_arrow_whitespace
shorthand_operator
single_test_class
sorted_first_last
sorted_imports
statement_position
strict_fileprivate
superfluous_disable_command
switch_case_alignment
switch_case_on_newline
syntactic_sugar
todo
trailing_closure
trailing_comma
trailing_newline
trailing_semicolon
trailing_whitespace
type_body_length
type_name
unavailable_function
unneeded_break_in_switch
unneeded_parentheses_in_closure_argument
untyped_error_in_catch
unused_closure_parameter
unused_enumerated
unused_optional_binding
valid_ibinspectable
vertical_parameter_alignment
vertical_parameter_alignment_on_call
vertical_whitespace
void_return
weak_delegate
xctfail_message
yoda_condition
Linting Swift files at paths 
Linting 'ViewController.swift' (1/4)
Linting 'UITests.swift' (2/4)
Linting 'AppDelegate.swift' (3/4)
Linting 'Tests.swift' (4/4)
/Users/daniel/path/to/project/AppDelegate.swift:30:50: warning: Discouraged Optional Collection Violation: Prefer empty collection over optional collection. (discouraged_optional_collection)
/Users/daniel/path/to/project/ViewController.swift:35:75: warning: Unused Closure Parameter Violation: Unused parameter "provider" in a closure should be replaced with _. (unused_closure_parameter)
/Users/daniel/path/to/project/ViewController.swift:33:7: warning: Trailing Closure Violation: Trailing closure syntax should be used whenever possible. (trailing_closure)
Done linting! Found 3 violations, 0 serious in 4 files.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
0.27.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
Homebrew
* Paste your configuration file:

```yml
disabled_rules: # rule identifiers to exclude from running

opt_in_rules: # some rules are only opt-in
  - attributes
  - closure_end_indentation
  - closure_spacing
  - collection_alignment
  - conditional_returns_on_newline
  - contains_over_first_not_nil
  - discouraged_optional_boolean
  - discouraged_optional_collection
  - empty_count
  - empty_string
  - explicit_enum_raw_value
  - explicit_init
  - fatal_error_message
  - first_where
  - function_default_parameter_at_end
  - implicit_return
  - multiline_arguments
  - multiline_function_chains
  - multiline_parameters
  - nimble_operator
  - operator_usage_whitespace
  - overridden_super_call
  - prohibited_super_call
  - redundant_nil_coalescing
  - redundant_type_annotation
  - redundant_void_return
  - sorted_first_last
  - sorted_imports
  - switch_case_on_newline
  - trailing_closure
  - unneeded_parentheses_in_closure_argument
  - unused_import
  - vertical_parameter_alignment_on_call
  - yoda_condition
  # Find all the available rules by running:
  # swiftlint rules

included: # paths to include during linting. `--path` is ignored if present.

excluded: # paths to ignore during linting. Takes precedence over `included`.
  - Carthage

analyzer_rules: # Rules run by `swiftlint analyze` (experimental)

# configurable rules can be customized from this configuration file
# binary rules can set their severity level
force_cast: warning # implicitly

force_try:
  severity: warning # explicitly

# rules that have both warning and error levels, can set just the warning level
# implicitly
# they can set both implicitly with an array
type_body_length:
  - 300 # warning
  - 400 # error

# or they can set both explicitly
file_length:
  warning: 500
  error: 1200

line_length:
  warning: 160
  error: 160

# naming rules can set warnings/errors for min_length and max_length
# additionally they can set excluded names
type_name:
  min_length: 4 # only warning
  max_length: # warning and error
    warning: 40
    error: 50
  excluded: # excluded via string array
    - dtiTests
    - dtiUITests

identifier_name:
  min_length: # only min_length
    error: 4 # only error
  excluded: # excluded via string array
    - id
    - url

reporter: "xcode" # reporter type (xcode, json, csv, checkstyle, junit, html, emoji, sonarqube)
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
No, not using nested configurations.
* Which Xcode version are you using (check `xcode-select -p`)?
Version 10.0 (10A255)
`xcode-select -p` --> xcode-select version 2349.
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// I would expect this to trigger the `collection_alignment` rule, but it does not:
import UIKit

class ViewController: UIViewController {
  override func viewDidLoad() {
    super.viewDidLoad()
    let abcd = [
      "alpha": "a",
        "beta": "b",
      "gamma": "g",
      "delta": "d",
        "epsilon": "e"
    ]
    print(abcd)
  }
}
```

**Comment by pranav21keshav** [comment]
I am facing this issue when using swift lint
 Loading configuration from '.swiftlint.yml'

 Configuration contains invalid keys:

 ["warning", "severity", "min_length", "error", "max_length"]

 Invalid configuration for 'identifier_name'. Falling back to default.

 Invalid configuration for 'type_name'. Falling back to default.

 Invalid configuration for 'file_length'. Falling back to default.

 Invalid configuration for 'force_try'. Falling back to default.


Following is my swiftlint.yml file



disabled_rules: # rule identifiers to exclude from running
- colon
- comma
- control_statement
opt_in_rules: # some rules are only opt-in
- empty_count
# Find all the available rules by running:
# swiftlint rules
included: # paths to include during linting. `--path` is ignored if present.
- VariousSchemes
excluded: # paths to ignore during linting. Takes precedence over `included`.
- Pods
analyzer_rules: # Rules run by `swiftlint analyze` (experimental)
- explicit_self

# configurable rules can be customized from this configuration file
# binary rules can set their severity level
force_cast: warning # implicitly
force_try:
severity: warning # explicitly
# rules that have both warning and error levels, can set just the warning level
# implicitly
line_length: 110
# they can set both implicitly with an array
type_body_length:
- 300 # warning
- 400 # error
# or they can set both explicitly
file_length:
warning: 500
error: 1200
# naming rules can set warnings/errors for min_length and max_length
# additionally they can set excluded names
type_name:
min_length: 4 # only warning
max_length: # warning and error
warning: 40
error: 50
excluded: iPhone # excluded via string
identifier_name:
min_length: # only min_length
error: 4 # only error
excluded: # excluded via string array
- id
- URL
- GlobalAPIKey
reporter: "xcode" # reporter type (xcode, json, csv, checkstyle, junit, html, emoji, sonarqube, markdown)

**Comment by pranav21keshav** [comment]
I got the answer.
It was because of the spacing issue when when specifying extra details for the rule. For example, **warning:** below, **file_length:**  should start with some blank space
